### PR TITLE
Allow relative file $ref links in OpenApi spec

### PR DESCRIPTION
**Pull request by damonhook**
## Description

- Create a new `add_spec_view_directory` directive
    - Allow you to split your openapi spec into multiple files (using the external file `$ref` links that came with OpenApi3).
    - You can test it by running the `app.py` located in `examples/splitfile`

Closes: #93

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -79,6 +79,24 @@ For responses, if the payload does not match the API document, an exception is r
 
 ## Advanced configuration
 
+### Relative File References in Spec
+
+A feature introduced in OpenApi3 is the ability to use `$ref` links to external files (https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#referenceObject).
+
+In order to use this you must ensure that you have all of your spec files in a given directory (ensure that you do not have any code in this directory as all files in it are exposed as static files), then **replace** the `pyramid_openapi3_spec` call that you did in [Getting Started](#getting-started) with the following:
```
Comment (sayanarijit) [inline README.md:86]:
```suggestion
In order to use this you must ensure that you have all of your spec files in a given directory (ensure that you do not have any code in this directory as all the files in it are exposed as static files), then **replace** the `pyramid_openapi3_spec` call that you did in [Getting Started](#getting-started) with the following:
```

**Review by sayanarijit — APPROVED** [review]

**Review by sayanarijit — COMMENTED** [review]
LGTM

**Comment by sayanarijit** [comment]
FYI missing coverage report will be visible in CircleCI artifacts if you rebase with master and push. Ref https://github.com/Pylons/pyramid_openapi3/commit/7f34ee1353d883d080323565fd6a7f1fe4cf9fb4

File: pyramid_openapi3/__init__.py [hunk 2]
```diff
@@ -200,6 +202,54 @@ def spec_view(request: Request) -> FileResponse:
...
+
+    :param filepath: absolute/relative path to the root specification file
+    :param route: URL path where to serve specification file
+    :param route_name: Route name under which specification file will be served
+    """
+
+    def register() -> None:
+        if route.endswith((".yaml", ".yml", ".json")):
+            logger.warning(
+                "Having be a filename is not recommended when using a spec directory"
```
Comment (zupo) [inline pyramid_openapi3/__init__.py:221]: I would raise an `Exception` here and say this is outright not allowed.

File: pyramid_openapi3/__init__.py [hunk 3]
```diff
@@ -200,6 +202,54 @@ def spec_view(request: Request) -> FileResponse:
...
+    route_name: str = "pyramid_openapi3.spec",
+) -> None:
+    """Serve and register OpenApi 3.0 specification directory.
+
+    :param filepath: absolute/relative path to the root specification file
+    :param route: URL path where to serve specification file
+    :param route_name: Route name under which specification file will be served
+    """
+
+    def register() -> None:
```
Comment (zupo) [inline pyramid_openapi3/__init__.py:218]: Can we have a check here that verifies that `add_spec_view` was not called and raises an error if it was?

**Review by zupo — COMMENTED** [review]
Looks good!

**Comment by zupo** [comment]
@damonhook: I specifically refrained from reviewing tests because you wrote it was a proof of concept. Ping me when tests are ready for reviewal.

**Comment by damonhook** [comment]
> @damonhook: I specifically refrained from reviewing tests because you wrote it was a proof of concept. Ping me when tests are ready for reviewal.

@zupo If you are happy with the approach then I think it is ready for a full reviewal. I have written the unit tests for it now as well

File: examples/splitfile/tests.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
+"""A couple of functional tests to showcase pyramid_openapi3 features."""
```
Comment (zupo) [inline examples/splitfile/tests.py:1]:
```suggestion
"""A couple of functional tests to showcase pyramid_openapi3 works with YAML spec split into multiple files."""
```

**Review by zupo — COMMENTED** [review]
Doc(string)s need a bit of polish, then it's good to merge!

File: examples/splitfile/README.md [hunk 5]
```diff
@@ -1,6 +1,6 @@
-# An example RESTful API app showcasing the power of pyramid_openapi3
+# An example RESTful API app showcasing how to split YAML spec into multiple files
 
-This folder showcases how to use the [pyramid_openapi3](https://github.com/Pylons/pyramid_openapi3) Pyramid add-on for building robust RESTful APIs. With only a few lines of code you get automatic validation of requests and responses against an OpenAPI v3 schema, along with Swagger "try-it-out" documentation for your API.
+This folder showcases how to use the [pyramid_openapi3](https://github.com/Pylons/pyramid_openapi3) Pyramid add-on for building robust RESTful APIs defined with a multi-file YAML schema.```
```
Comment (zupo) [inline examples/splitfile/README.md]:
```suggestion
This folder showcases how to use the [pyramid_openapi3](https://github.com/Pylons/pyramid_openapi3) Pyramid add-on for building robust RESTful APIs defined with a multi-file YAML schema.
```

**Comment by zupo** [comment]
Thanks @damonhook!

**Pull request merged**